bundles/specmate-migration-test/src/com/specmate/migration/test/AddAttributeTest.java

```diff
@@ -0,0 +1,229 @@
+
+public class AddAttributeTest {
+	private static BundleContext context;
+	private static ServiceController<CDOPersistencyService> persistencyServiceController;
+	private static ServiceController<BaselineModelProviderImpl> baselineModelController;
+	private static ServiceController<AttributeAddedModelProviderImpl> attributeAddedModelController;
+	private static ConfigurationAdmin configurationAdmin;
+	private static IPersistencyService persistencyService;
+	private static IMigratorService migratorService;
+	
+	@BeforeClass
+	public static void init() throws Exception {
+		context = FrameworkUtil.getBundle(AddAttributeTest.class).getBundleContext();
+		
+		configurationAdmin = getConfigurationAdmin();
+		Dictionary<String, Object> properties = new Hashtable<>();
+		properties.put(CDOPersistenceConfig.KEY_JDBC_CONNECTION, "jdbc:h2:mem:testdb;DB_CLOSE_DELAY=-1");
+		//properties.put(CDOPersistenceConfig.KEY_JDBC_CONNECTION, "jdbc:h2:./database/specmate");
+		properties.put(CDOPersistenceConfig.KEY_REPOSITORY_NAME, "testrepo");
+		properties.put(CDOPersistenceConfig.KEY_RESOURCE_NAME, "r1");
+		properties.put(CDOPersistenceConfig.KEY_USER_RESOURCE_NAME, "r2");	
+		configurationAdmin.getConfiguration(CDOPersistenceConfig.PID).update(properties);
+		
+		baselineModelController = new ServiceController<>(context);
+		baselineModelController.register(IPackageProvider.class, BaselineModelProviderImpl.class, null);
+		attributeAddedModelController = new ServiceController<>(context);
+		attributeAddedModelController.register(IPackageProvider.class, AttributeAddedModelProviderImpl.class, null);
+		persistencyServiceController = new ServiceController<>(context); 
+		migratorService = getMigratorService();
+				
+		activatePersistency(baselineModelController.getService());
+		addBaselinedata();
+		deactivatePersistency();
+	}
+	
+	@AfterClass
+	public static void shutdown() {
+		baselineModelController.unregister();
+		attributeAddedModelController.unregister();
+	}
+	
+	private static void activatePersistency(IPackageProvider p) throws Exception {
+		persistencyServiceController.register(IPersistencyService.class, CDOPersistencyService.class, null);
+		persistencyService = persistencyServiceController.getService();
+		CDOPersistencyService cdop = (CDOPersistencyService) persistencyService;
+		startSupportServices(cdop, p);
+		cdop.activate();
+	}
+	
+	private static void deactivatePersistency() {
+		CDOPersistencyService cdop = (CDOPersistencyService) persistencyService;
+		cdop.deactivate();
+		persistencyServiceController.unregister();
+	}
+	
+	@Test 
+	public void testNeedsMigration() throws Exception {
+		activatePersistency(baselineModelController.getService());
+		assertFalse(migratorService.needsMigration());
+		migratorService.setModelProviderService(attributeAddedModelController.getService());
+		assertTrue(migratorService.needsMigration());
+		deactivatePersistency();
+	}
+	
+	@Test
+	public void doMigration() throws Exception {
```

Comment:
for readability reasons you could break this method up in several methods

Reply:
Yes, agree.